Sources/SwiftProtobuf/ProtobufBinaryDecoding.swift

```diff
@@ -682,14 +674,16 @@ private class ProtobufScanner {
 
     // Parse index/type marker that starts each field.
     // This also bookmarks the start of field for a possible skip().
-    func getTagType() throws -> Int? {
+    func getTag() throws -> FieldTag? {
         fieldStartP = p
         fieldStartAvailable = available
         if let t = try getRawVarint() {
             if t > 7 && t < UInt64(UInt32.max) {
```

Comment:
Done. I had to keep the `t < UInt64(UInt32.max)` case because it would require `FieldTag` to accept larger than 32-bit values, which I would consider a design flaw. A better way to handle that would be to have the scanner use a separate `getRawVarint32` method, as other language implementations do, and have *that* throw if the value encountered is actually larger than 32-bits.